src/HonzaBotner.Discord.Services/Managers/ButtonManager.cs

```diff
@@ -0,0 +1,56 @@
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using DSharpPlus;
+using DSharpPlus.Entities;
+using DSharpPlus.EventArgs;
+using HonzaBotner.Discord.Managers;
+using HonzaBotner.Discord.Services.Options;
+using Microsoft.Extensions.Options;
+
+namespace HonzaBotner.Discord.Services.EventHandlers
+{
+    public class ButtonManager : IButtonManager
+    {
+        private readonly CommonCommandOptions _config;
+        private readonly DiscordConfig _dcConfig;
+
+        public ButtonManager(IOptions<CommonCommandOptions> options, IOptions<DiscordConfig> config)
+        {
+            _config = options.Value;
+            _dcConfig = config.Value;
+
+        }
+
+        public async Task SetupButtons(GuildDownloadCompletedEventArgs eventArgs)
+        {
+            if (eventArgs.Guilds == null) return;
+
+            DiscordGuild guild;
+
+            try
+            {
+                guild = eventArgs.Guilds[_dcConfig.GuildId ?? 0];
+            }
+            catch (KeyNotFoundException)
+            {
+                return;
+            }
+
+            DiscordChannel channel = guild.GetChannel(_config.VerificationChannelId);
+            if (channel.Type != ChannelType.Text) return;
+
+            DiscordMessage message = await channel.GetMessageAsync(_config.VerificationMessageId);
+
+            var builder = new DiscordMessageBuilder()
```

Comment:
How will this work with restarting the client, will it remove old buttons and add new one?

Reply:
Buttons itself will be removed and re-added but those buttons have same IDs so even in case some desktop/mobile user is somehow late on updates from server, it will still send same command. Only critical moment is when the bot is off. Otherwise it should cause no harm.

Reply:
Buttons use developer-provided buttons as opposed to slash commands and other interactions, which are generated on the Discord side, so should be fine from my understanding

Reply:
I don't understand the point of this. Are buttons really removed so we need to initialize this step every Run? From my tests, it's not true.

Rather than doing this, we should make some commands to be able to add/update buttons to a message, and that generator should do it for us. As Jirka says, buttons keep developer-provided IDs, so it's fine just handling those, no need to remove them and create them again, hmm?

Reply:
Will work on button setup and remove command. How should I handle config values for verification messages? Should I have simple command like ’button setup verification’ and it will automagically read config values and add buttons or should I remove that from config and do smth like ’button setup verification %url%’?

Reply:
This comment was addressed. Buttons are now assigned using `message buttons setup %target_message%` command.